### PR TITLE
Hide RoundState Implementation

### PR DESCRIPTION
**Pull request by mcortesi**
### Description

Hide `RoundState` and `MessageSet` behind interface to avoid bad usage. For example, access data to read/write without the proper lock setup

### Tested

unit tests

### Other changes

minor lint issues

### CODE REVIEW

File: consensus/istanbul/core/message_set.go [hunk 1]
```diff
@@ -36,13 +36,26 @@ func newMessageSet(valSet istanbul.ValidatorSet) *messageSet {
...
+	fmt.Stringer
+	Add(msg *istanbul.Message) error
+	GetAddressIndex(addr common.Address) (uint64, error)
+	GetAddressPublicKey(addr common.Address) ([]byte, error)
+	ValSetSize() uint64
+	Remove(address common.Address)
+	Values() (result []*istanbul.Message)
+	Size() int
+	Get(addr common.Address) *istanbul.Message
+	// String() string
```
Comment (asaj) [inline consensus/istanbul/core/message_set.go]: Why is this commented out?

File: consensus/istanbul/core/roundstate.go [hunk 2]
```diff
@@ -87,88 +125,107 @@ func (s *roundState) Subject() *istanbul.Subject {
...
 			Sequence: new(big.Int).Set(s.sequence),
 		},
-		Digest: s.Preprepare.Proposal.Hash(),
+		Digest: s.preprepare.Proposal.Hash(),
 	}
 }
 
-func (s *roundState) SetPreprepare(preprepare *istanbul.Preprepare) {
+func (s *roundStateImpl) Preprepare() *istanbul.Preprepare {
+	s.mu.RLock()
```
Comment (asaj) [inline consensus/istanbul/core/roundstate.go]: Why does this have locking but commits, prepares, and parentcommits don't?

File: consensus/istanbul/core/roundstate.go [hunk 3]
```diff
@@ -177,22 +234,34 @@ func (s *roundState) CreateAndSetPreparedCertificate(quorumSize int) error {
...
 	return nil
 }
 
+func (s *roundStateImpl) PreparedCertificate() istanbul.PreparedCertificate {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+	return s.preparedCertificate
+}
+
+func (s *roundStateImpl) SetPreparedCertificate(preparedCertificate istanbul.PreparedCertificate) {
```
Comment (asaj) [inline consensus/istanbul/core/roundstate.go]: Is this only used for testing? If so, should we remove it?

File: consensus/istanbul/core/message_set.go (hunk 1 above)
Comment (mcortesi) [reply consensus/istanbul/core/message_set.go]: oh, that's the Stringer interface 
i'll remove the comment

File: consensus/istanbul/core/roundstate.go (hunk 3 above)
Comment (mcortesi) [reply consensus/istanbul/core/roundstate.go]: yes

**Pull request merged**